### PR TITLE
realesrgan-ncnn-vulkan: fix build failure

### LINKED ISSUE #289910: Build failure: Real-ESRGAN-ncnn-vulkan
### Steps To Reproduce

1. build Real-ESRGAN-ncnn-vulkan-0.2.0
2. See error

### Build log

```
nix { "action": "setPhase", "phase": "unpackPhase" }
Running phase: unpackPhase
unpacking source archive /nix/store/qghd7p96y55wqvym8nq4fb56pwrhis4w-source
source root is source/src
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: patchPhase
applying patch /nix/store/7mxx0p4lkw3h4kg5470da3nzma93h7zs-cmakelists.patch
patching file CMakeLists.txt
applying patch /nix/store/ac63jyzkan3ivar36i7v4jhr70pqx724-models_path.patch
patching file main.cpp
substituteStream(): WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file 'main.cpp')
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: configurePhase
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/n>
CMake Warning (dev) at CMakeLists.txt:17 (project):
  cmake_minimum_required() should be called prior to this top-level project()
  call.  Please see the cmake-commands(7) manual for usage documentation of
  both commands.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- The C compiler identification is GNU 13.2.0
-- The CXX compiler identification is GNU 13.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/4cjqvbp1jbkps185wl8qnbjpf8bdy8j9-gcc-wrapper-13.2.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/4cjqvbp1jbkps185wl8qnbjpf8bdy8j9-gcc-wrapper-13.2.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Found OpenMP_C: -fopenmp (found version "4.5")
-- Found OpenMP_CXX: -fopenmp (found version "4.5")
-- Found OpenMP: TRUE (found version "4.5")
-- Found Vulkan: /nix/store/ji07cn67ysm5b2i1fxmd6crgs7iks6qa-vulkan-loader-1.3.275.0/lib/libvulkan.so (found version "1.3.275") found components: glslangValidator missing components: glslc
-- Found glslangValidator: /nix/store/c9h11l07ldmkdss7jkd6r11sf49kn72w-glslang-14.0.0/bin/glslangValidator
-- Using glslang install located at /nix/store/c9h11l07ldmkdss7jkd6r11sf49kn72w-glslang-14.0.0/lib/cmake
CMake Warning at /nix/store/c9h11l07ldmkdss7jkd6r11sf49kn72w-glslang-14.0.0/lib/cmake/OSDependentTargets.cmake:2 (message):
  Using `OSDependentTargets.cmake` is deprecated: use `find_package(glslang)`
  to find glslang CMake targets.
Call Stack (most recent call first):
  CMakeLists.txt:112 (include)


CMake Error at CMakeLists.txt:113 (include):
  include could not find requested file:

    /nix/store/c9h11l07ldmkdss7jkd6r11sf49kn72w-glslang-14.0.0/lib/cmake/OGLCompilerTargets.cmake


CMake Warning at /nix/store/c9h11l07ldmkdss7jkd6r11sf49kn72w-glslang-14.0.0/lib/cmake/glslangTargets.cmake:2 (message):
  Using `glslangTargets.cmake` is deprecated: use `find_package(glslang)` to
  find glslang CMake targets.
Call Stack (most recent call first):
  CMakeLists.txt:118 (include)


CMake Warning at /nix/store/c9h11l07ldmkdss7jkd6r11sf49kn72w-glslang-14.0.0/lib/cmake/SPIRVTargets.cmake:2 (message):
  Using `SPIRVTargets.cmake` is deprecated: use `find_package(glslang)` to
  find glslang CMake targets.
Call Stack (most recent call first):
  CMakeLists.txt:119 (include)


-- Found OpenMP_C: -fopenmp (found version "4.5")
-- Found OpenMP_CXX: -fopenmp (found version "4.5")
-- Found WebP: /nix/store/76d3zi6hl8zrim5k4936g1xa651fh9rr-libwebp-1.3.2/include
-- Configuring incomplete, errors occurred!
```

### Additional context

None I can think of

### Notify maintainers

@tilcreator @GetPsyched @bjornfor @emilytrau @h7x4

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.7.4, NixOS, 24.05 (Uakari), 24.05.20240215.a4d4fe8`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"home-manager, nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by iynaix**
fixes #289910

Updates build for glslang 14.0.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/graphics/realesrgan-ncnn-vulkan/default.nix [hunk 1]
```diff
@@ -50,15 +51,19 @@ stdenv.mkDerivation rec {
 
   installPhase = ''
     mkdir -p $out/bin $out/share
+
     cp realesrgan-ncnn-vulkan $out/bin/
+    wrapProgram $out/bin/realesrgan-ncnn-vulkan \
+      --prefix LD_LIBRARY_PATH ':' "${lib.makeLibraryPath [ vulkan-loader ]}"
```
Comment (SuperSandro2000) [inline pkgs/tools/graphics/realesrgan-ncnn-vulkan/default.nix:57]: Can we use patchelf instead? LD_LIBRARY_PATH has many disadvantages and taints spawned programs from the binary.

**Comment by LamprosPitsillos** [comment]
Result of `nixpkgs-review pr 294207` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>realesrgan-ncnn-vulkan</li>
  </ul>
</details>

File: pkgs/tools/graphics/realesrgan-ncnn-vulkan/default.nix [hunk 2]
```diff
@@ -50,15 +50,22 @@ stdenv.mkDerivation rec {
...
     mkdir -p $out/bin $out/share
+
     cp realesrgan-ncnn-vulkan $out/bin/
     cp -r ${models}/models $out/share
   '';
 
+  postFixup = ''
+    patchelf --add-rpath ${lib.makeLibraryPath [ vulkan-loader ]} \
+      --add-needed libvulkan.so \
+      $out/bin/realesrgan-ncnn-vulkan
```
Comment (SuperSandro2000) [inline pkgs/tools/graphics/realesrgan-ncnn-vulkan/default.nix:61]:
```suggestion
    patchelf $out/bin/realesrgan-ncnn-vulkan --add-needed libvulkan.so
```
I think we just need this

Comment (iynaix) [reply pkgs/tools/graphics/realesrgan-ncnn-vulkan/default.nix]: Yep, that works! Made the change

**Pull request merged**